### DISCUSSION TITLE
SwiftUI Example

### DISCUSSION THREAD

**Discussion by eSuba** [sbertix/Swiftagram#238] [Q&A] [answered]
Hi !
Is it possible to use Swiftagram with SwiftUI ? It's not specified in the doc.
It would be really great if someone can share me an exemple of it. Logging in using swiftui and get basic infos like username, followers count etc.

Thanks 🙏

**Comment by sbertix** [comment, marked as answer]
Hey @eSuba,

Check out the examples in the appropriately named "Examples" directory: they both use SwiftUI 😊

**Reply by eSuba** [reply]
Hey @sbertix,

Thanks for your fast answer ! 😁
Based on the example "Followers" I want to add a display of some infos like username, following/followers count, posts counts etc. Would you mind helping me a bit please
I have some trouble using Swiftagram 😅, I tried something like this but it obviously didn't work.
```
  @Published private(set) var summ: User?

        secret.removeDuplicates { $0?.identifier == $1?.identifier }
            .flatMap { secret -> AnyPublisher<[User]?, Never> in
                guard let secret = secret else { return Just(nil).eraseToAnyPublisher() }
                
                return Endpoint.user(secret.identifier)
                .summary
                .unlock(with: secret)
                .session(.instagram)
                .catch { _ in Just(nil) }
                .eraseToAnyPublisher()
            }
                .receive(on: RunLoop.main)
                .assign(to: \.summ, on: self)
                .store(in: &bin)
```

**Reply by sbertix** [reply]
What your code should do is every time you log in with a different account and push the `Secret` to `secret`, assuming it's a valid relay, it fetches the logged in user's info and assign them to `summ`, in your case.
But it doesn't cause you're somehow expecting an optional array of `User`s to be outputted at one point, which is not the case. 
So it seems like your intent differs immensely from the code presented here. 

I'm sorry I can't help you further: discussions here are for **Swiftagram** related doubts/questions only, and this appears to be more of either a lack of **Combine** understanding or uncertainties about the library you can quickly settle either through the docs and the `README.md`, or even just analyzing the _Followers_ example.
